airbyte-integrations/connectors/source-exact/source_exact/streams.py

```diff
@@ -0,0 +1,465 @@
+
+                if not pendulum.parse(cursor_value).timezone_name in ["UTC", "+00:00"]:
+                    self.logger.warning(f"The value of the cursor field 'Modified' is not detected as a UTC timestamp: {cursor_value}. This might lead to an incorrect $filter clause and unexpected records. ")
+
+                tz_cet = pendulum.timezone("CET")
+                utc_timestamp = pendulum.parse(cursor_value)
+                cet_timestamp = tz_cet.convert(utc_timestamp)
+                cet_timestamp_str = cet_timestamp.isoformat().split("+")[0]
+
+                params["$filter"] = f"Modified gt datetime'{cet_timestamp_str}'"
+            else:
+                raise RuntimeError(f"Source not capable of incremental syncing with cursor field '{self.cursor_field}'")
+
+        if self.cursor_field == "Modified":
+            params["$orderby"] = "Modified asc"
+
+        return params
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        """
+        If response contains the __next property, there are more pages. This property contains the full url to
+        call next including endpoint and all query parameters.
+        """
+
+        response_json = response.json()
+        next_url = response_json.get("d", {}).get("__next")
+
+        return {"next_url": next_url} if next_url else None
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """Overridden to parse results from the nested array."""
+
+        response_json = response.json()
+        results = response_json.get("d", {}).get("results", [])
+
+        return [self._parse_item(x) for x in results]
+
+    def read_records(self, sync_mode: SyncMode, stream_slice: Mapping[str, Any] = None, **kwargs) -> Iterable[StreamData]:
+        """Overridden to change the url_base based on the current division, and to keep track of the cursor."""
+
+        division = str(stream_slice["division"])
+        self._url_base = f"https://start.exactonline.nl/api/v1/{division}/"
+
+        self.logger.info(f"Syncing division {division}...")
+
+        # Reset state for full refresh
+        if sync_mode == SyncMode.full_refresh:
+            self._state_per_division[division] = {}
+
+        for record in super().read_records(sync_mode=sync_mode, stream_slice=stream_slice, **kwargs):
+            # Track the largest cursor value
+            if self.cursor_field and sync_mode == SyncMode.incremental:
+                cursor_value = record[self.cursor_field]
+                current_cursor_value = self._state_per_division[division].get(self.cursor_field)
+                current_cursor_value = cursor_value if not current_cursor_value else current_cursor_value
+
+                if current_cursor_value:
+                    self._state_per_division[division].update({self.cursor_field: max(cursor_value, current_cursor_value)})
+
+            yield record
```

Comment:
Can you explain why you're overriding the base function?

Reply:
It was documented in the pydoc, but I refactored it a bit I hope it is clearer now why its needed.